packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -164,19 +189,27 @@ const PipelineWrapper: React.FC<IProps> = ({
       .then((images: any) => {
         runtimeImages.current = images;
         const nodesCopy = JSON.parse(JSON.stringify(nodes));
+        const pipelinePropertiesCopy = JSON.parse(
+          JSON.stringify(updatedPipelineProperties)
+        );
         for (const node of nodesCopy) {
           node.properties.uihints.parameter_info[1].data.items = Object.values(
             runtimeImages.current
           );
         }
+        pipelinePropertiesCopy.properties.uihints.parameter_info[4].data.items = Object.values(
+          runtimeImages.current
+        );
         setUpdatedNodes(nodesCopy);
+        setUpdatedPipelineProperties(pipelinePropertiesCopy);
         changeHandler();
       })
       .catch(error => RequestErrors.serverError(error));
 
     return (): void => {
       currentContext.model.contentChanged.disconnect(changeHandler);
     };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We shouldn't ignore these warnings

Reply:
It caused an infinite loop when I added that dependency - what would you suggest as the solution for that?

Reply:
This should be addressed in a follow-up PR that adds support for default runtime image - for now, this issue doesn't need to be addressed yet.